### ISSUE TITLE
[FR] Support API to dynamically add fine-tuned models to Mlflow server

### ISSUE THREAD

**Issue by ys-1234** [mlflow/mlflow#11785]
### Willingness to contribute

No. I cannot contribute this feature at this time.

### Proposal Summary

Currently, the MLflow Deployments Server relies on a user-provided configuration file, written in YAML, that defines the endpoints and providers available to the server (https://mlflow.org/docs/latest/llms/deployments/index.html#configuring-the-deployments-server). Whenever there is a new model, the configuration file needs to be updated to reflect the change on deployment server.
However, the solution could be hard to scale when people develop multiple new fine-tuned models in parallel. Multiple people might try to update the same configuration file at the same time. 
Instead of solely relying on updating the configuration file to add new models, it will be nice to have an API to dynamically add new models to the deployment server.

### Motivation

> #### What is the use case for this feature?
When people develop multiple new fine-tuned models in parallel, they want to deploy the models to Mlflow deployment server smoothly.
> #### Why is this use case valuable to support for MLflow users in general?

> #### Why is this use case valuable to support for your project(s) or organization?

> #### Why is it currently difficult to achieve this use case?

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by BenWilson2** [comment]
Hi @ys-1234 this is a fantastic idea! We'll open this up for community contribution! :) Thank you for letting us know of this pain point :)

**Comment by martinbomio** [comment]
@BenWilson2 I was looking into support for this recently. Great that we have this issue to track that work.
Thinking about an implementation for this feature, I wonder if we could introduce the concept of `models_provider` or `models` (or probably some better naming to avoid confusion with the existing provider) to the config, where a endpoint can be queried to get a list of models. For example, OpenAI already supports the concept of a `/models` endpoint that could be used here. I am sure other providers have something similar

So the config could look like this:
```
      - name: llama2-7b
        endpoint_type: llm/v1/chat
        models:
          provider: openai
          config:
            openai_api_key: ....
```

**Comment by zhouyou9505** [comment]
Hi sir, I am interested in this. Could you assign it to me?

**Comment by RUTUPARNk** [comment]
Hi, I’m interested in contributing! Is this issue open to first-timers? Is there any design or prior discussion for how this dynamic model API should work?
